keps/sig-node/2837-pod-level-resource-spec/README.md

```diff
@@ -1172,6 +1207,179 @@ back to aggregating container requests.
+      memory: 200Mi
+  containers:
+    - name: c1
+      image: registry.k8s.io/pause:latest
+      resizePolicy:
+        - resourceName: "cpu"
+          restartPolicy: "NotRequired"
+        - resourceName: "memory"
+          restartPolicy: "RestartRequired"
+    - name: c2
+      image: registry.k8s.io/pause:latest
+      resources:
+        requests:
+          cpu: 50m
+          memory: 50Mi
+        limits:
+          cpu: 100m
+          memory: 100Mi
+      resizePolicy:
+        - resourceName: "cpu"
+          restartPolicy: "NotRequired"
+        - resourceName: "memory"
+          restartPolicy: "RestartRequired"
+```
+
+In this example:
+* CPU resizes: Neither container requires a restart for CPU resizes, and therefore CPU resizes at neither the container nor pod level will trigger any restarts.
+* Container c1 (inherited memory limit): c1 does not define any container level
+  resources, so the effective memory limit of the container is determined by the
+  pod-level limit. When the pod's limit is resized, c1's effective memory limit
+  changes. Because c1's memory resizePolicy is RestartRequired, a resize of the
+  pod-level memory limit will trigger a restart of container c1.
+* Container c2 (specified memory limit): c2 does define container-level resources,
+  so the effective memory limit of c2 is the container level limit. Therefore, a
+  resize of the pod-level memory limit doesn't change the effective container limit,
+  so the c2 is not restarted when the pod-level memory limit is resized.
+
+##### Implementation Details
+
+###### Allocating Pod-level Resources
+Allocation of pod-level resources will work the same as container-level resources. The allocated resources checkpoint will be extended to include pod-level resources, and the pod object will be updated with the allocated resources in the pod sync loop.
+
+###### Actuating Pod-level Resource Resize
+A dirty bit for pod-level resources will be added to the allocation checkpoint to
+signal the need for resize.  This change, along with the existing tracking and
+triggering mechanisms, is sufficient for pod-level resize actuation.  The same
+ordering rules for pod and container resource resizing will be applied for each
+resource as needed:
+1. Increase pod-level cgroup (if needed)
+2. Decrease container resources
+3. Decrease pod-level cgroup (if needed)
+4. Increase container resources
+
+###### Tracking Actual Pod-level Resources
+To accurately track actual pod-level resources during in-place pod resizing, several
+changes are required:
+
+1. Configuration reading: Pod-level resource config is currently read as part of the
+   resize flow, but will also need to be read during pod creation. Critically, the
+   configuration must be read again after the resize operation to capture the
+   updated resource values.  Currently, the configuration is only read before a
+   resize.
+   
+2. Pod Status Update: Because the pod status is updated before the resize takes
```

Comment:
Aside: We should probably re-evaluate this, outside the context of this KEP. Now that there are things reflected in the status that don't also trigger resync, we're going to need to resync the pod just to write another field to the status. I'm not sure off hand what the consequences of moving the status update to the end of PodSync would be.